packages/ckeditor5-ui/src/panel/balloon/balloonpanelview.ts

```diff
@@ -387,6 +395,27 @@ export default class BalloonPanelView extends View {
 		this.listenTo( global.window, 'resize', () => {
 			this.attachTo( options );
 		} );
+
+		// Hide the panel if the target element is no longer visible.
+		if ( targetElement && !this._resizeObserver ) {
+			const checkVisibility = () => {
+				// If the target element is no longer visible, hide the panel.
+				if ( !targetElement.checkVisibility() ) {
```

Comment:
AFAIK we have a helper for `isVisible()`. I'm not sure how this is different from this DOM method. Could you please verify it?

Reply:
It works exactly like this one above, I replaced it with your suggestion to make it more convenient.